include/pybind11/cast.h

```diff
@@ -837,6 +864,57 @@ template <typename T> object cast(const T &value,
 template <typename T> T handle::cast() const { return pybind11::cast<T>(*this); }
 template <> inline void handle::cast() const { return; }
 
+template <typename T>
+typename std::enable_if<detail::move_always<T>::value || detail::move_if_unreferenced<T>::value, T>::type move(object &&obj) {
```

Comment:
I think the enable_if can be removed here. The function will only ever be instantiated if move_always or move_if_unreferenced are true.